src/input/manager.rs

```diff
@@ -196,13 +197,49 @@ impl Manager {
 
     /// Starts listening for [Command] messages to be sent from clients and
     /// dispatch those events.
-    pub async fn run(&mut self) -> Result<(), Box<dyn Error>> {
-        // Start tasks for discovering new input devices
-        self.watch_input_devices().await?;
+    pub async fn run(&mut self) -> Result<(), Box<dyn Error + Send + Sync>> {
+        let tx_for_listen_on_dbus = self.tx.clone();
+        let dbus_for_listen_on_dbus = self.dbus.clone();
 
-        // Create a DBus interface
-        self.listen_on_dbus().await?;
+        let cmd_tx_all_devices = self.tx.clone();
+        let cmd_tx_iio = self.tx.clone();
+        let cmd_tx_hidraw = self.tx.clone();
 
+        // Watch for hidraw/evdev inotify events.
+        // TODO: when we reload the udev device it triggers the udev watcher. We do this to break
+        // access to the file descriptor for processes that have already authenticated. Figure out
+        // a way to do this only using the udev events.
+        let (watcher_tx, mut watcher_rx) = mpsc::channel(BUFFER_SIZE);
+        if std::path::Path::new(DEV_PATH).exists() {
+            let tx = watcher_tx.clone();
+            tokio::task::spawn_blocking(move || {
+                log::info!("Started hidraw device discovery thread");
+                watcher::watch(DEV_PATH.into(), tx)
+            });
+        }
+        if std::path::Path::new(INPUT_PATH).exists() {
+            let tx = watcher_tx.clone();
+            tokio::task::spawn_blocking(move || {
+                log::info!("Started evdev device discovery thread");
+                watcher::watch(INPUT_PATH.into(), tx)
+            });
+        }
+
+        log::debug!("Starting input manager task...");
+
+        let _ = tokio::join!(
+            Self::discover_all_devices(&cmd_tx_all_devices),
+            Self::watch_iio_devices(cmd_tx_iio),
+            Self::watch_devnodes(cmd_tx_hidraw, &mut watcher_rx),
```

Comment:
Should probably change the name of the cloned tx here since the func name changed.

Reply:
Done by replacing with self.tx.clone()